docs/blog/_posts/2019-04-05-14th-dotty-milestone-release.md

```diff
@@ -0,0 +1,200 @@
+
+```scala
+opaque type IArray[T] = Array[T]
+
+object IArray {
+
+  implied arrayOps {
+    inline def (arr: IArray[T]) apply[T] (n: Int): T = (arr: Array[T]).apply(n)
+    inline def (arr: IArray[T]) length[T] : Int = (arr: Array[T]).length
+  }
+  def apply[T: ClassTag](xs: T*): IArray[T] = Array(xs: _*)
+  /*...*/
+}
+```
+
+Essentially, the above defines a wrapper around the ordinary `Array` which exposes only its `apply` (to get an element by index) and `length` methods:
+
+- `opaque type IArray[T]` defines a type which is known to be an `Array`, but this information is known only in its companion object `IArray`. To the rest of the world, this information is not available.
+- The `implied arrayOps` implied instance defines the extension methods that expose the operations available on the `IArray` type.
+- The extension methods, `apply` and `length`, delegate to these of the `Array` type. These methods are inlined which means the performance footprint is the same as that of the original `Array` methods.
+- Because the methods are `inline` and because the `IArray` type is `opaque`, we need the `(arr: Array[T])` casts. `IArray` is known to be equal to `Array` only inside the `IArray` object and the inlining will cause the extension methods' bodies to appear outside the `IArray` object when these methods are called.
+
+## Creator Applications
+This new language feature is a generalisation of the ability to construct case classes without the `new` keyword:
+
+```scala
+class StringBuilder(s: String) {
+  def this() = this("")
+}
+StringBuilder("abc")  // same as new StringBuilder("")
+StringBuilder()       // same as new StringBuilder()
+```
+
+The motivation for the feature is mainly ergonomic. To make it possible, a new interpretation was added to a function call `f(a)`. Previously, the rules were as follows:
+
+Given a function call `f(args)`,
+
+ - if `f` is a method applicable to `args`, typecheck `f(args)` unchanged,
+ - otherwise, if `f` has an `apply` method applicable to `args` as a member, continue with `f.apply(args)`,
+ - otherwise, if `f` is of the form `p.m` and there is an implicit conversion `c` applicable to `p` so that `c(p).m` is applicable to `args`, continue with  `c(p).m(args)`
+
+ There's now a fourth rule following these rules:
+
+ - otherwise, if `f` is syntactically a stable identifier, and `new f` where `f` is interpreted as a type identifier is applicable to `args`, continue with `new f(args)`.
+
+For more information, please see the [documentation](http://dotty.epfl.ch/docs/reference/other-new-features/creator-applications.html).
+
+## Other changes
+
+Some of the other changes include:
+
+- `infer` method renamed to `the`, the semantics of which is now the same as that of the `the` method of Shapeless. Namely, the implicits are resolved more precisely – see this [gist](https://gist.github.com/milessabin/8833a1dbf7e8245b30f8) for an example in Shapeless.

```

Comment:
I would put this too --> http://dotty.epfl.ch/docs/reference/contextual/inferable-params.html#querying-implied-instances

Reply:
Included